controllers/controlplane_controller_reconciler_utils.go

```diff
@@ -248,3 +249,18 @@ func (r *ControlPlaneReconciler) ensureClusterRoleBindingForControlPlane(
 	labelObjForControlPlane(clusterRoleBinding)
 	return true, clusterRoleBinding, r.Client.Create(ctx, clusterRoleBinding)
 }
+
+func (r *ControlPlaneReconciler) ensureCertificate(
+	ctx context.Context,
+	controlplane *operatorv1alpha1.ControlPlane,
+) (bool, string, error) {
+	secretName := controlplane.Name + "-control-mtls-cert"
+	usages := []certificatesv1beta1.KeyUsage{certificatesv1beta1.UsageKeyEncipherment,
+		certificatesv1beta1.UsageDigitalSignature, certificatesv1beta1.UsageClientAuth}
+	// this subject is arbitrary. data planes only care that client certificates are signed by the trusted CA, and will
+	// accept a certificate with any subject
+	created, err := maybeCreateCertificateSecret(ctx, fmt.Sprintf("%s.%s", controlplane.Name, controlplane.Namespace),
```

Comment:
nit: "maybe" do something can be a bit vague, but if what we're really doing is either GET or PUT, perhaps:

```suggestion
	created, err := getOrCreateCertificateSecret(ctx, fmt.Sprintf("%s.%s", controlplane.Name, controlplane.Namespace),
```

Reply:
We've used it elsewhere, e.g. https://github.com/Kong/kubernetes-ingress-controller/blob/4a7d27aa4be6bde04b916197b9bdefdc034231f2/internal/manager/controllerdef.go#L48, I think Michal had suggested the convention for functions that may not take action. It's not actually getting anything, just indicating if it did create the certificate, so maybe sounds more appropriate, WDYT?

Reply:
Sure, and good point. Not a blocker.